src/Module.ts

```diff
@@ -798,6 +798,11 @@ export default class Module {
 		resolvedIds?: ResolvedIdMap;
 		transformFiles?: EmittedFile[] | undefined;
 	}): void {
+		/* eslint-disable-next-line unicorn/number-literal-case */
+		if (code.charCodeAt(0) === 0xfe_ff) {
+			code = code.slice(1);
+		}
+
```

Comment:
Reading up in Wikipedia about BOMs, I am not 100% sure if this fully resolves the issue in all cases, but it looks reasonable.

Reply:
Yeah, I'm also not 100% sure, but I had a look at `SWC` and `strip-bom`(npm package), It seems that they also do the same thing. If there are any uncovered cases, we can fix them later.